dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java

```diff
@@ -671,38 +671,62 @@ public void overridesMultitreesByPersonalization(final String pageId,
 
         if (!multiTrees.isEmpty()) {
 
-            final List<Params> insertParams = Lists.newArrayList();
-
-            for (final MultiTree tree : multiTrees) {
-                //This is for checking if the content we are trying to add is already added into the container
-                db.setSQL(SELECT_COUNT_MULTI_TREE_BY_RELATION_PERSONALIZATION_PAGE_CONTAINER_AND_CHILD)
-                        .addParam(tree.getRelationType())
-                        .addParam(tree.getPersonalization())
-                        .addParam(pageId)
-                        .addParam(tree.getContainerAsID())
-                        .addParam(tree.getContentlet())
-                        .addParam(tree.getVariantId());
-                final int contentExist = Integer.parseInt(db.loadObjectResults().get(0).get("cc").toString());
-                if(contentExist != 0){
-                    final String contentletTitle = APILocator.getContentletAPI().findContentletByIdentifierAnyLanguage(tree.getContentlet()).getTitle();
-                    final String errorMsg = String.format("Content '%s' [ %s ] has already been added to Container " +
-                                                                  "'%s'", contentletTitle, tree.getContentlet(),
-                            tree.getContainer());
-                    Logger.debug(this,errorMsg);
-                    throw new IllegalArgumentException(errorMsg);
-                }
-
-                insertParams
-                        .add(new Params(pageId, tree.getContainerAsID(), tree.getContentlet(),
-                                tree.getRelationType(), tree.getTreeOrder(), tree.getPersonalization(), tree.getVariantId()));
-            }
-            db.executeBatch(INSERT_SQL, insertParams);
+            copyMultiTree(pageId, multiTrees);
         }
         this.refreshContentletReferenceCount(originalContentletIds, multiTrees);
         updateHTMLPageVersionTS(pageId, variantId);
         refreshPageInCache(pageId, variantId);
     }
 
+    public void copyMultiTree(final String pageId, final List<MultiTree> multiTrees) throws DotDataException {
+        copyMultiTree(pageId, multiTrees, null);
+    }
+
+
+    /**
+     * Copy a collection of {@link MultiTree} but to a different {@link Variant}.
+     *
+     * @param pageId {@link String} Page's identifier
+     * @param multiTrees {@link List} of {@link MultiTree} to copy
+     * @param variantName {@link String} name of the variant to copy to
+     * @throws DotDataException
+     */
+    @Override
+    public void copyMultiTree(final String pageId, final List<MultiTree> multiTrees,
```

Comment:
missing wrap in tranx

Reply:
done https://github.com/dotCMS/core/pull/24734/commits/0f1a100df316aa0d1e1af45ed5f4d56cf5cf7efa#diff-926e9e9e22e2c282fa305bbe50022e93d9d91d6b83a5c33c8e29cacde4a0b52cR696